src/edu/wright/cs/raiderplanner/controller/MenuController.java

```diff
@@ -431,6 +438,82 @@ public void handle(TouchEvent event) {
 		this.mainContent.addRow(2, moduleBox);
 	}
 
+	/**
+	 * Handles when the user selects the new profile button on the main screen.
+	 */
+	public void createNewProfile() {
+		MainController.save();
+		File plannerFile = null;
+		try {
+			Account newAccount = MainController.ui.createAccount();
+			StudyPlannerController study = new StudyPlannerController(newAccount);
+			// Welcome notification:
+			Notification not = new Notification("Welcome!", new GregorianCalendar(),
+					"Thank you for using RaiderPlanner!");
+			study.getPlanner().addNotification(not);
+			MainController.setSpc(study);
+			plannerFile = MainController.ui.savePlannerFileDialog();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		if (plannerFile != null) {
+			if (plannerFile.getParentFile().exists()) {
+				if (plannerFile.getParentFile().canRead()) {
+					if (plannerFile.getParentFile().canWrite()) {
+						MainController.setPlannerFile(plannerFile);
+						MainController.save();
+					} else {
+						UiManager.reportError("Directory can not be written to.");
+					}
+				} else {
+					UiManager.reportError("Directory cannot be read from.");
+				}
+
+			} else {
+				UiManager.reportError("Directory does not exist.");
+			}
+		}
```

Comment:
You may want to move this if block into a function, similar to `openProfile`. It looks like you have already written this code for a second time.